choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerSessionDBService.java

```diff
@@ -1,237 +1,302 @@
+			final Optional<ChoreographerSession> sessionOpt = sessionRepository.findById(sessionId);
+			if (sessionOpt.isEmpty()) {
+				worklogAndThrow("Session step registration has been failed", new InvalidParameterException("Session with id " + sessionId + " not exists"));
+			}
+			final ChoreographerSession session = sessionOpt.get();
+			
+			final Optional<ChoreographerStep> stepOpt = stepRepository.findById(stepId);
+			if (stepOpt.isEmpty()) {
+				worklogAndThrow(session.getPlan().getName(), session.getId(), "Session step registration has been failed", new InvalidParameterException("Step with id " + stepId + " not exists"));
+			}
+			final ChoreographerStep step = stepOpt.get();
+			
+			final Optional<ChoreographerExecutor> executorOpt = executorRepository.findById(executorId);
+			if (executorOpt.isEmpty()) {
+				worklogAndThrow(session.getPlan().getName(), session.getId(), "Session step registration has been failed", new InvalidParameterException("Executor with id " + executorId + " not exists"));
+			}
+			final ChoreographerExecutor executor = executorOpt.get();
+			
+			final ChoreographerSessionStep sessionStep = sessionStepRepository.saveAndFlush(new ChoreographerSessionStep(session, step, executor, status, message.trim()))	;
+			worklog(session.getPlan().getName(), step.getAction().getName(), step.getName(), session.getId(), "New session step has been registrated with id " + sessionStep.getId(), null);
+			return sessionStep;
+        	
+        } catch (final InvalidParameterException ex) {
+        	throw ex;
+        	
+        } catch (final Exception ex) {
+        	logger.debug(ex.getMessage(), ex);
+        	throw new ArrowheadException(CoreCommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
+        }
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    @Transactional(rollbackFor = ArrowheadException.class)
+    public ChoreographerSessionStep changeSessionStepStatus(final long sessionStepId, final ChoreographerSessionStepStatus status, final String message) {
+    	logger.debug("changeSessionStepStatus started...");
+        Assert.notNull(status, "ChoreographerSessionStepStatus is null");
+        Assert.isTrue(!Utilities.isEmpty(message), "message is empty");
+        
+        try {
+			final Optional<ChoreographerSessionStep> optional = sessionStepRepository.findById(sessionStepId);
+			if (optional.isEmpty()) {
+				worklogAndThrow("Session step status change has been failed", new InvalidParameterException("Session step with id " + sessionStepId + " not exists"));
+			}
+			ChoreographerSessionStep sessionStep = optional.get();
+			sessionStep.setStatus(status);
+			sessionStep.setMessage(message.trim());
+			
+			sessionStep = sessionStepRepository.saveAndFlush(sessionStep);
+			worklog(sessionStep.getSession().getPlan().getName(), sessionStep.getStep().getAction().getName(), sessionStep.getStep().getName(), sessionStep.getStep().getId(),
```

Comment:
move sessionStep.getStep() to a local variable to reduce the length of this line